### DISCUSSION TITLE
Expanding an Expander without resizing the entire panel?

### DISCUSSION THREAD

**Discussion by AQDuck97** [AvaloniaUI/Avalonia#15383] [Q&A] [answered]
I have an expander in a wrappanel and when I expand the expander I would like only the header be within the parent panel and have the expanded body flow over.
I've been unable to find anything about that, closest thing being ZIndex problems but that's not what this is.

I want to be able to use this program on Steam Deck as well and the way GameScope handles windows makes flyouts a no-go as they will be treated as a new window.

With combobox I can sort of achieve this, but that will also select the checkbox item.

Instead of this:...
![image](https://github.com/AvaloniaUI/Avalonia/assets/102743075/058fa4a4-3e15-4583-9fcc-ad6233e668bf)

...I would like this:
![image](https://github.com/AvaloniaUI/Avalonia/assets/102743075/ff64f11b-a789-4440-a44a-5b3303b0c24f)

**Comment by robloo** [comment]
It isn't possible to do what you are trying to do with these controls (without retemplating). You need to use a control that overlays on top of others rather than something embedded in the parent panel.

To do this you need to use a flyout/popup that opens when the button/header is pressed. This is how ComboBox works internally.

I would recommend in this case just to use the DropDownButton which is designed to simplify this case. If you want the chevron that rotates you will have to do more work. At that point it may be possible to re-template the expander itself and show content in a flyout.

**Reply by AQDuck97** [reply]
I was able to workaround it by simply adding a new grid so that it would be stacked on top

**Comment by AQDuck97** [comment, marked as answer]
I found that I could just make another base grid grid and put both the list below and the expander on the same grid and that worked great!

```xaml
<Grid>
    <Grid>
        <!-- the list -->
    </Grid>
    <Expander>
        
    </Expander>
</Grid>
```

![image](https://github.com/AvaloniaUI/Avalonia/assets/102743075/7b6875b5-a9ac-49c0-80c9-08139609b996)
